client/src/components/students-form/StudentsForm.tsx

```diff
@@ -0,0 +1,106 @@
+import Paper from '@mui/material/Paper';
+import ListSubheader from '@mui/material/ListSubheader';
+import List from '@mui/material/List';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemText from '@mui/material/ListItemText';
+import Checkbox from '@mui/material/Checkbox';
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import CardActions from '@mui/material/CardContent';
+// import { SubmitStudentsFormButton } from './SubmitStudentsFormButton';
+import { useEffect, useState } from 'react';
+import { fetchTerms } from '../../api/api-utils';
+import React from 'react';
+import { Day, Term } from '../../types/types';
+import { SubmitStudentsFormButton } from './SubmitStudentsFormButton';
+
+const compareTerms = (x: Term, term: Term) => {
+    return x.dayOfWeek === term.dayOfWeek &&
+        x.startTime === term.startTime &&
+        x.endTime === term.endTime;
+}
+
+interface TermsByDayGroup {
+    dayOfWeek: Day,
+    terms: Term[]
+}
+
+const groupTermsByDay = (terms: Term[]) => {
```

Comment:
const groupTermsByDay = (terms: Term[]) => {
  const termsByDayGroups: { [day in Day]: Term[] } = terms.reduce((acc, term) => {
    acc[term.dayOfWeek].push(term);
    return acc;
  }, { Monday: [], Tuesday: [], Wednesday: [], Thursday: [], Friday: [] });

  return Object.entries(termsByDayGroups).map(([dayOfWeek, terms]) => ({ dayOfWeek: dayOfWeek as Day, terms }));
};

Reply:
lubie pisane funkcyjnie ale jest git, taka ciekawostka

Reply:
Tez lubie funkcyjnie, ale uznalem ze postawie na czytelnosc kodu, ale dziekuje za tip, wiem juz jak indeksowac po enumie :)